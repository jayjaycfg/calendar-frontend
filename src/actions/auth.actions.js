import Swal from "sweetalert2";
import {types} from '../types/types';
import {fetchToken, fetchWithToken} from "../helpers/fetch";
import {SWAL_TEXT} from "../helpers/text";

const login =(user)=>({
    type : types.authLogin,
    payload : user
});

export const startLogin =(email,password)=>{
    return async (dispatch)=>{
        const response = await fetchToken('auth',{email,password},'POST');
        const {msg,name,ok,token,uid} = await response.json();
        if(ok){
            localStorage.setItem('token', token);
            localStorage.setItem('token-init-date', new Date().getTime());
            dispatch(login({
                uid,
                name
            }));
        }else{
            await Swal.fire(SWAL_TEXT.ERROR_TITLE, msg);
        }
    };
};

const register = (user)=>({
    type: types.authStartRegister,
    payload : user
});

export const startRegister=(email,name,password)=>{
    return async (dispatch)=>{
        const response = await fetchToken('auth/new',{email,name,password},'POST');
        const {msg,name : nameU,ok,token,uid} = await response.json();
        if(ok){
            localStorage.setItem('token',token);
            localStorage.setItem('token-init-date',new Date().getTime());
            dispatch(register({
                uid,
                nameU
            }));
        }else{
            await Swal.fire(SWAL_TEXT.ERROR_TITLE,msg);
        }

    };
};

const checking = ()=>({
    type : types.authCheckingFinish
});

export const startChecking=()=>{
    return async (dispatch)=>{
        const response = await fetchWithToken('auth/renew');
        const {name,ok,token,uid} = await response.json();

        if(ok){
            localStorage.setItem('token',token);
            localStorage.setItem('token-init-date',new Date().getTime());
            dispatch(login({
                uid,
                name
            }));
        }else{
            dispatch(checking());
        }
    };
};

const logout =()=>({type : types.authLogout});

export const startLogout =()=>{
    return (dispatch)=>{
        localStorage.clear();
        dispatch(logout());
    };
};