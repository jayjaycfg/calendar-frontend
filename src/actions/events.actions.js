import Swal from 'sweetalert2';
import {types} from '../types/types';
import {fetchWithToken} from "../helpers/fetch";
import prepareEvent from "../helpers/prepareEvents";
import {SWAL_TEXT} from "../helpers/text";


const addNewEvent=(event)=>({type: types.eventAddNew, payload: event});

export const startAddNewEvent =(event)=>{
    return async (dispatch, getState)=>{
        const {uid, name} = getState().auth;
        try{
            const response = await fetchWithToken('events',event,'POST');
            const {ok,eventoGuardado} = await response.json();

            if(ok){
                event.id = eventoGuardado.id;
                event.user = {
                    _id : uid,
                    name
                };

                dispatch(addNewEvent(event));
            }else{
                return Swal.fire(SWAL_TEXT.ERROR_TITLE,SWAL_TEXT.ERROR_MESSAGE,SWAL_TEXT.ERROR_ICON);
            }
        }catch (e) {
          return Swal.fire(SWAL_TEXT.ERROR_TITLE,e,SWAL_TEXT.ERROR_ICON);
        }
    };
};

export const setActiveEvent=(event)=>({type: types.eventSetActive, payload: event});
export const clearActiveEvent=()=>({type: types.eventClearActive});

const updateEvent=(event)=>({type: types.eventUpdate, payload: event});

export const startEventUpdate = (event)=>{

    return async (dispatch)=>{
        try{
            const response = await fetchWithToken(`events/${event.id}`,event,'PUT');
            const {msg,ok} = await response.json();
            if(ok){
                dispatch(updateEvent(event));
            }else{
                return Swal.fire(SWAL_TEXT.ERROR_TITLE,msg,SWAL_TEXT.ERROR_ICON);
            }
        }catch (e) {
            return Swal.fire(SWAL_TEXT.ERROR_TITLE,e,SWAL_TEXT.ERROR_ICON);
        }
    };
};

const deleteEvent=()=>({type: types.eventDelete});

export const startEventDelete =()=>{
    return async (dispatch,getState)=>{
        const {id} = getState().calendar.activeEvent;
        try{
            const response = await fetchWithToken(`events/${id}`,undefined,'DELETE');
            const {msg,ok} = await response.json();
            if(ok){
               dispatch(deleteEvent());
            }else{
                return Swal.fire(SWAL_TEXT.ERROR_TITLE,msg,SWAL_TEXT.ERROR_ICON);
            }
        }catch (e) {
            return Swal.fire(SWAL_TEXT.ERROR_TITLE,e,SWAL_TEXT.ERROR_ICON);
        }
    };
};

const eventLoaded =(events)=>({type: types.eventLoaded, payload : events});

export const startEventLoading =()=>{
    return async (dispatch)=>{
        try{
            const response = await fetchWithToken('events');
            const {ok, eventsBD} = await response.json();

            if(ok){
                const events = prepareEvent(eventsBD);
                dispatch(eventLoaded(events));
            }else{
                return Swal.fire(SWAL_TEXT.ERROR_TITLE,SWAL_TEXT.ERROR_MESSAGE,SWAL_TEXT.ERROR_ICON);
            }

        }catch (e) {
            return  Swal.fire(SWAL_TEXT.ERROR_TITLE,e,SWAL_TEXT.ERROR_ICON);
        }
    };
};