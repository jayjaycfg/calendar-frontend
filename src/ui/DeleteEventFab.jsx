import React from 'react';
import {useDispatch} from 'react-redux';
import {startEventDelete} from "../actions/events.actions";
import {FIELD_TEXT} from "../helpers/text";

const DeleteEventFab = () => {
    const dispatch = useDispatch();
    return (
        <button
            className='btn btn-danger fab-danger'
            onClick={()=>dispatch(startEventDelete())}
        >
            <i className='fas fa-trash'/>
            <span>{FIELD_TEXT.BTN_DELETE}</span>
        </button>
    );
};

export default DeleteEventFab;