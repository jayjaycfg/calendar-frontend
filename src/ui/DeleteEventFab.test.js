import React from "react";
import {mount} from "enzyme";
import thunk from "redux-thunk";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import DeleteEventFab from "./DeleteEventFab";
import {startEventDelete} from "../actions/events.actions";

jest.mock("../actions/events.actions",()=>({
    startEventDelete : jest.fn()
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialState = {};
let store = mockStore(initialState);
store.dispatch = jest.fn();

let wrapper = mount(
    <Provider store={store}>
        <DeleteEventFab/>
    </Provider>
);
describe('<DeleteEventFab/> Component', () => {
    test('should ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should delete an event', async() => {
        const btn = wrapper.find('button');
        btn.simulate('click');
        expect(store.dispatch).toHaveBeenCalledTimes(1);
        expect(startEventDelete).toHaveBeenCalledTimes(1);
    });
});