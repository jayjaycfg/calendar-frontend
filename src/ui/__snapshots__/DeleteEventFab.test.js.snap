// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteEventFab/> Component should  1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
     Borrar evento
  </span>
</button>
`;
